win32/devdoc/threadpool_win32.md

```diff
@@ -243,3 +249,63 @@ static VOID CALLBACK on_timer_callback(PTP_CALLBACK_INSTANCE instance, PVOID con
 **SRS_THREADPOOL_WIN32_42_017: [** Otherwise `context` shall be used as the context created in `threadpool_schedule_work`. **]**
 
 **SRS_THREADPOOL_WIN32_42_018: [** The `work_function` callback passed to `threadpool_schedule_work` shall be called, passing to it the `work_function_context` argument passed to `threadpool_schedule_work`. **]**
+
+```c
+static VOID CALLBACK on_work_callback_v2(PTP_CALLBACK_INSTANCE instance, PVOID context, PTP_TIMER timer);
+```
+
+`on_work_callback_v2` executes the work function passed to `threadpool_create_work_item`.
+
+**SRS_THREADPOOL_WIN32_05_001: [** If `context` is `NULL`, `on_work_callback_v2` shall return. **]**
+
+**SRS_THREADPOOL_WIN32_05_002: [** Otherwise `context` shall be used as the context created in `threadpool_create_work_item`. **]**
+
+**SRS_THREADPOOL_WIN32_05_003: [** The `work_function` callback passed to `threadpool_create_work_item` shall be called with the `work_function_context` as an argument. `work_function_context` was set inside the `threadpool_create_work_item` as an argument to `CreateThreadpoolContext`. **]**
+
+```c
+MOCKABLE_FUNCTION(, PVOID, threadpool_create_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, PVOID, work_function_context);
+```
+
+`threadpool_create_work_item` creates a work item to be executed by the threadpool.
+
+**SRS_THREADPOOL_WIN32_05_004: [** `threadpool_create_work_item` shall initialize the `work_item_context` to `NULL`, where `NULL` would be a failure indicator to the calling functions **]**
+
+**SRS_THREADPOOL_WIN32_05_005: [** If `threadpool` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**SRS_THREADPOOL_WIN32_05_006: [** If `work_function` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**SRS_THREADPOOL_WIN32_05_007: [** Otherwise `threadpool_create_work_item` shall allocate a context where `work_function`, `work_function_context`, and `ptp_work` shall be saved. **]**
+
+**SRS_THREADPOOL_WIN32_05_008: [** If any error occurs, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**SRS_THREADPOOL_WIN32_05_009: [** `threadpool_create_work_item` shall create `PTP_WORK` by calling `CreateThreadpoolWork` to set the callback function as `on_work_callback_v2` and pass the newly created context whenever the callback is called in the tp_environment. **]**
```

Comment:
>and pass the newly created context whenever the callback is called in the tp_environment [](http://example.com/codeflow?start=177&length=88)

I don't think we need this part of the spec ("and pass...") because that is 05_003